lhotse/utils.py

```diff
@@ -30,6 +31,69 @@
 _lhotse_uuid: Optional[Callable] = None
 
 
+class SmartOpen:
+    """Wrapper class around smart_open.open method
+
+    The smart_open.open attributes are cached as classed attributes - they play the role of singleton pattern.
+
+    The SmartOpen.setup method is intended for initial setup.
+    It imports the `open` method from the optional `smart_open` Python package,
+    and sets the parameters which are shared between all calls of the `smart_open.open` method.
+
+    If you do not call the setup method it is called automatically in SmartOpen.open with the provided parameters.
+
+    The example demonstrates that instantiating S3 `session.client` once,
+    instead using the defaults and leaving the smart_open creating it every time
+    has dramatic performance benefits of 44s vs 18.9s Wall time.
+
+    import boto3
+    session = boto3.Session()
+    client = session.client('s3')
+    from lhotse.utils import SmartOpen
+
+    if not slow:  # switch between 44s vs 18.9 Wall time
```

Comment:
Sorry to be nit-picking -- this 44s vs 18.9s seems very specific to your use-case, could we remove it?

Reply:
Also -- can you format it like:

```
Example::
    >>> import boto3
    >>> session = boto3.Session()
    ...
```

This will render well with sphinx.

Reply:
Sure!

Reply:
Can you pls check that it is ok now?

Or just tell me how to generate (and check) the docs myself.

Reply:
The auto generated docs for this PR are here https://lhotse--335.org.readthedocs.build/en/335/

RTD used to add it as a CI test result in GH but the integration broke when I moved Lhotse to lhotse-speech organization… and I couldn’t fix it. 

Unfortunately it seems that the API reference doesn’t list lhotse.utils so you won’t see it there anyway.. I haven’t gotten to re-working this part of the docs yet.